Physically remove the folder src/coreclr/tests

trylek opened the pull request:
This change removes the last few files from src/coreclr/tests
under src/tests and finally depletes the folder. One change of note
is rename of src/coreclr/tests/runtest.py to src/tests/run.py
in accordance with the previous renames.

I have put the runparallel.sh script to src/tests as it seems
to have general usability, I moved the rest of the scripts to
src/tests/Common/scripts. I haven't actually tried to make it
work. I managed to catch one reference from arm32_ci_script /
arm_ci_test.sh to bringup_runtest.sh that I fixed but again
without actually testing them.

I am not too familiar with these scripts, I haven't found
references to them anywhere in the infra, some of them seem
related to the JIT team, adding Bruce to the PR to confirm.
I have no idea whether they are used at all, any insights into
what obsolete scripts we can get rid of is more than welcome.

Thanks

Tomas

/cc: @dotnet/runtime-infrastructure

trylek committed:
Physically remove the folder src/coreclr/tests

This change removes the last few files from src/coreclr/tests
under src/tests and finally depletes the folder. One change of note
is rename of src/coreclr/tests/runtest.py to src/tests/run.py
in accordance with the previous renames.

I have put the runparallel.sh script to src/tests as it seems
to have general usability, I moved the rest of the scripts to
src/tests/Common/scripts. I haven't actually tried to make it
work. I managed to catch one reference from arm32_ci_script /
arm_ci_test.sh to bringup_runtest.sh that I fixed but again
without actually testing them.

I am not too familiar with these scripts, I haven't found
references to them anywhere in the infra, some of them seem
related to the JIT team, adding Bruce to the PR to confirm.
I have no idea whether they are used at all, any insights into
what obsolete scripts we can get rid of is more than welcome.

Thanks

Tomas

BruceForstall commented on src/tests/run.py:
```diff
@@ -66,7 +66,7 @@
 
 # Import coreclr_arguments.py from src\coreclr\scripts
-sys.path.append(os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "scripts"))
+sys.path.append(os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "coreclr", "scripts"))
```
There are a few references to "runtest.py" in this script that should be updated to match the renaming.

trylek replied:
Updated in 2nd commit, thanks for reminding me!

BruceForstall commented on src/tests/run.cmd:
```diff
@@ -176,7 +176,7 @@ if defined RunInUnloadableContext (
 )
 
-set NEXTCMD=python "%__RepoRootDir%\src\coreclr\tests\runtest.py" %__RuntestPyArgs%
+set NEXTCMD=python "%__RepoRootDir%\src\tests\run.py" %__RuntestPyArgs%
```
fwiw, I wish this script was a much smaller wrapper around run.py. It looks like Jarret left a lot of stuff here because the Python version isn't used in the official build scenario. It would be nice to understand why that needs to be different, or if it was just left different because it was easier to leave it.

trylek replied:
I plan to look into both pairs of scripts (build.cmd / sh, run.cmd / sh) and start trimming them down with the ultimate goal to make most test-related logic directly runnable from the root build script per https://github.com/dotnet/runtime/issues/33183.

ViktorHofer replied:
FWIW I would prefer to remove dependencies on python if possible. It's just another build dependency that we don't depend on in the libraries and installer part of the repo. Just my 5c.

trylek replied:
I have already witnessed several lengthy and somewhat controversial discussions on the subject. I'm no big fan of Python myself but at the very least I want to separate that discussion from the mechanical task of removing the <code>src/coreclr/tests</code> folder. This PR should conclude the shuffle I started in the July infra rollout and my expectation is that, as the next step, I'll drill deeper into the details of the build and run scripts with the ultimate goal to make tests runnable from the root build script as I mentioned. I hope that, at least in this case, there'll be no Python involved and the root build logic will directly call into the MSBuild projects <code>src/tests/build.proj</code> and <code>src/tests/run.proj</code>.

ViktorHofer replied:
Yeah sorry for bringing this topic which is mostly unrelated, up here. I understand that higher up in the stack the python dependency is less an issue (like dotnet/performance) but in dotnet/runtime we might want to rethink if we want to depend on it. What you are saying is great, I'm a big fan of common paths with least amount of dependencies.

BruceForstall reviewed (approved):
You can just delete protoonjit_testenv.cmd, x64_arm64_altjit.cmd, and x86_arm_altjit.cmd -- these can be useful for JIT devs, but were there to support some jobs in Jenkins that we no longer have.

trylek committed:
Address Bruce's PR feedback

I have deleted the obsolete scripts protononjit_testenv,
x64_arm64_altjit and x86_arm_altjit, and I have updated references
to runtest.py / runtest.proj in the renamed run.py script. For
making run.cmd / run.sh thinner wrappers, I'll tackle that in a
follow-up change, I plan to do a similar thing for build.cmd / sh.

Thanks

Tomas
